[Khairul Aizat] iP

kairuler opened the pull request.

kairuler committed:
Added Level-0: Greet

kairuler committed:
Renamed the chatbot to "Bro"

kairuler committed:
Added Greet, Echo, Exit

kairuler committed:
Added: Bro echoes commands entered by user

kairuler committed:
Added: Add, List

kairuler committed:
Added: Mark as Done + Mark as Not Done

kairuler committed:
Refactor to fit coding standard

Remove magic strings, instead use named constants
Use more variables when referencing class methods to avoid long lines of code and improve readability

kairuler committed:
Add ToDos, Events, Deadlines

Rayleigh47 commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,110 @@
+                break;
+            case "mark":
+                int indexOfMarkedTask = Integer.parseInt(words[1]) - 1;
+                boolean isValidated1 = validateTask(tasks.size(), indexOfMarkedTask);
```
Naming for isValidated1 and isValidated2 can be more specific, for eg. isValidatedMark, is ValidatedUnmark.

Rayleigh47 commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,110 @@
+     * @param taskIndex Index of Task being validated
+     * @return true if Task is valid, false if Task is invalid
+     */
+    private static boolean validateTask(int tasks, int taskIndex) {
```
Boolean methods can be phrased as a question, for example isATask.

kairuler replied:
Thanks! I've renamed the boolean method to `isAValidTask`.

Rayleigh47 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,50 @@
+    public void setType(String type) {
+        this.type = type;
+    }
+
```
Maybe use a different method name markOrNot, to make it clearer

Rayleigh47 commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,110 @@
+                for (int i = indexOfEndTime + 1; i < words.length; ++i) {
+                    endTime.append(" ").append(words[i]);
+                }
+                Task event = new Event(eventName.toString().trim(), startTime.toString().trim(), endTime.toString().trim());
```
Can consider splitting the line to keep it shorter.

Rayleigh47 commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,110 @@
+        StringBuilder reply = new StringBuilder();  // Use StringBuilder as we concatenate Strings in a loop later on
+        Scanner in = new Scanner(System.in);
+        ArrayList<Task> tasks = new ArrayList<>();  // Dynamic array to store text entered by user
+        boolean endInput = false;
```
Use boolean method names like isInput instead.

Rayleigh47 reviewed (commented):
Comments on coding standard violations

MichelleLiang0116 commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,110 @@
+            line = in.nextLine();
+            String[] words = line.split(" ");
+            switch (words[0]) {
+            case "bye":
```
Probably should consider case sensitive.

MichelleLiang0116 commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,110 @@
+                tasks.add(event);
+                reply = new StringBuilder(" added: " + event);
+                break;
+            }
```
Can add default to avoid improper input.

kairuler replied:
Thank you! I've added a `default` block in case the user inputs an invalid command.

MichelleLiang0116 commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,110 @@
+public class Bro {
+    public static final String HORIZONTAL_LINE = "\n───────────────────────────────────────────────────────────────\n";
+    public static final String GREETING = " Sup bro. I'm Bro.\n" + " What do you want?";
+    public static final String TASK_DOES_NOT_EXIST = " Bro that task number does not exist...";
```
Good to use constant variable for reuse

MichelleLiang0116 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+    private String deadline;
+    public Deadline(String name, String deadline) {
+        super(name);
+        this.setType("D");
```
Good to use setType.

MichelleLiang0116 reviewed (commented):
Overall is good
Using constant variable and more convenient command.

TayJiunYuan commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,110 @@
+        boolean endInput = false;
+        while (!endInput) {
+            line = in.nextLine();
+            String[] words = line.split(" ");
```
Perhaps use a more specific term rather than "words"

TayJiunYuan commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,110 @@
+                StringBuilder endTime = new StringBuilder();
+                int indexOfStartTime = Arrays.asList(words).indexOf("/from");
+                int indexOfEndTime = Arrays.asList(words).indexOf("/to");
+                for (int i = 1; i < indexOfStartTime; ++i) {
```
Perhaps comment what this for loop does so that it is easier to follow.

TayJiunYuan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,50 @@
+    private String name;
+    private boolean completed;
+    private String type;
+
```
Perhaps keep the line spacings consistent, ie. there are line spacings in some files but none in others.

TayJiunYuan commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,110 @@
+            case "list":
+                reply = new StringBuilder(" Your tasks:\n");
+                for (int i = 0; i < tasks.size(); ++i) {
+                    Task task = tasks.get(i);
```
Perhaps use currentTask instead of task as it is confusing between the object task and class Task.

TayJiunYuan reviewed (commented):
Overall readable and organised code, good job!

kairuler committed:
Handle input error for "Mark/Unmark as done" feature

kairuler committed:
Handle input error for "Deadline"

kairuler committed:
Handle input error for "Todo"

kairuler committed:
Merge branch 'branch-Level-5'

kairuler committed:
Add Tasks and Exceptions packages

kairuler committed:
Merge branch 'branch-A-Packages'

kairuler committed:
Add enum "Type"

Enum "Type" represents the different query types e.g. mark, todo, event. Currently its only use is in exception handling of the input.

kairuler committed:
Add Delete

kairuler committed:
Add Save

kairuler committed:
Merge branch 'branch-Level-6'

kairuler committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Bro.java

kairuler committed:
Package the App as a JAR file

kairuler committed:
Fix behaviour of Save when file is not found

mathanmahe commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,219 @@
+import bro.tasks.ToDo;
+import bro.tasks.Deadline;
+import bro.tasks.Event;
+public class Bro {
```
Java coding standard:  Logical units within a block should be separated by one blank line.

Consider leaving a line between import statements and class definition.

mathanmahe commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,219 @@
+import bro.tasks.Deadline;
+import bro.tasks.Event;
+public class Bro {
+    public static final String HORIZONTAL_LINE = "\n─────────────────────────────────────────────────────────────────────────────────────────────\n";
```
Java coding standard:  Line length should be no longer than 120 chars.

You can consider breaking down the string using line wrapping.

mathanmahe commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,219 @@
+    static final String PATH_NAME = "saved_tasks.txt";
+    public static final String GREETING = " Sup bro. I'm Bro.\n" + " What do you want?";
+    public static final String TASK_DOES_NOT_EXIST = " Bro that task number does not exist...";
+//    static final String FILE_NOT_FOUND = "File not found";
```
Java coding standard: Comments should be indented relative to their position in the code. 

Do ensure that the comment has the same level of indent as the code surrounding it.

mathanmahe commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,219 @@
+        System.out.println(HORIZONTAL_LINE + GREETING + HORIZONTAL_LINE);
+        // User Input
+        String line;
+        StringBuilder reply;  // Use StringBuilder as we concatenate Strings in a loop later on
```
Java coding standard: COmments should be indented relative to their position in the code. 

Consider adding this to the previous line before declaration of `StringBuilder reply`

mathanmahe commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,219 @@
+                break;
+            case "mark": // Fallthrough
+            case "unmark":
+                boolean markAsComplete = arrayOfInputs[0].equals("mark");   // this boolean decides if the following `markComplete()` marks the task as Completed or Uncompleted
```
Java coding standard: Boolean should be named to sound like booleans. 
Can consider renaming it to `isMarkedAsComplete` or `isMarkedTask`

mathanmahe commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,219 @@
+                for (int i = 0; i < taskList.size(); ++i) {
+                    Task currentTask = taskList.get(i);
+                    String mark = currentTask.mark();
+                    reply.append(" ").append(i + 1).append(".[").append(currentTask.getType()).append("][").append(mark).append("] ").append(currentTask).append("\n");
```
As per the coding standard, do break down long lines like this across all occurences in the code.

mathanmahe commented on src/main/java/Bro.java:
```diff
@@ -0,0 +1,219 @@
+    /**
+     * Validate the input format, creates a Todo object and returns Bro's reply.
+     *
+     * @param taskList List of all tasks
```
Good job on writing header comments on private methods!

mathanmahe commented on src/main/java/bro/exceptions/invalidInputFormat.java:
```diff
@@ -0,0 +1,44 @@
+
+import bro.tasks.Type;
+
+public class invalidInputFormat extends Exception {
```
Java coding standard: Class name should be in PascalCase.

`InvalidInputFormat`

mathanmahe reviewed (commented):
Good job bro. Some improvements to Coding Standard as well as Code quality have been highlighted in my review

kairuler committed:
Commit from Laptop

kairuler committed:
Merge branch 'master' of https://github.com/kairuler/ip

# Conflicts:
#	src/main/java/Bro.java
#	src/main/java/Save.java

kairuler committed:
Add More OOP

kairuler committed:
Add Find feature

kairuler committed:
Add JavaDoc comments

kairuler committed:
Merge pull request #1 from kairuler/branch-Level-9

Add Find feature

kairuler committed:
Merge pull request #2 from kairuler/branch-A-JavaDoc

Add JavaDoc comments

kairuler committed:
Merge branch 'master' of https://github.com/kairuler/ip

kairuler committed:
Add User Guide

kairuler committed:
Release v2.0
